component/query.libsonnet

```diff
@@ -44,9 +47,101 @@ local query = thanos.query(queryBaseConfig + params.commonConfig + params.query
       type: params.query.serviceType,
     },
   },
+  serviceAccount+: {
+    metadata+: {
+      annotations+: if params.queryRbacProxy.enabled then {
+        'serviceaccounts.openshift.io/oauth-redirecturi.primary': params.queryRbacProxy.redirectUri,
+        // there is also another annotation:
+        //  serviceaccounts.openshift.io/oauth-redirectreference.primary: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"<route-name>"}}'
+        // However, using an Ingress object generates a randomized Route name, and setting `kind: Ingress` didn't work.
+        // So we have to give an static URL.
+      },
+    },
+  },
+  deployment+: {
+    spec+: {
+      template+: {
+        spec+: {
+          containers+: if params.queryRbacProxy.enabled then [
+            {
+              name: 'openshift-oauth-proxy',
+              image: '%s/%s:%s' % [ proxyImage.registry, proxyImage.image, proxyImage.tag ],
+              args: [
+                '--upstream=http://0.0.0.0:9090',
+                '--provider=openshift',
+                '--cookie-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token',
+                '--openshift-service-account=%s' % 'thanos-query',
+                // TODO: Replace JSON below with `std.manifestJsonMinified({...}})` once available in newer jsonnet version
+                // Only cluster-admins should be able to get system resources
+                '--openshift-sar={"namespace":"%s","resource":"services","name":"%s","verb":"get"}' % [ params.namespace, params.queryRbacProxy.serviceName ],
+                '--http-address=http://:%s' % params.queryRbacProxy.port,
+                '--https-address=',
```

Comment:
I wouldn't spend the effort for re-encrypted traffic, since that would most likely be yet another OpenShift-specific feature in a component which should work on any K8s distribution